addons/stock/models/stock_lot.py

```diff
@@ -128,14 +128,15 @@ def _compute_delivery_ids(self):
     @api.depends('quant_ids')
     def _compute_single_location(self):
         for lot in self:
-            location_id = lot.quant_ids.filtered(lambda q: q.quantity > 0).mapped('location_id')
-            lot.location_id = location_id if len(location_id) == 1 else False
+            quant_ids = lot.quant_ids.filtered(lambda q: q.quantity > 0)
+            lot.location_id = quant_ids.location_id if len(quant_ids) == 1 else False
```

Comment:
note to self: chech with using len() on location_id instead
potential bug when concurrency issue on writing quants